docs/src/specifications.md

```diff
@@ -0,0 +1,207 @@
+# Problem Specifications
```

Comment:
In general we should try to avoid large blocks of Julia code in the docs.  It is too burdensome to keep the docs up to date with the actual implementation.  Providing doc strings in-line with the implementation is a more maintainable approach.